systemd-cryptsetup-generator doesn't build for Systemd 237

## Issue description

`nix-build -A systemd-cryptsetup-generator` fails with the following error since eb862c48ddc288243c5447e8decc623b10ef165c#diff-dd830f049b8e42facd9261dc425529f1:

```
Build targets in project: 1425
Found ninja-1.8.2 at /nix/store/d6xvh2izw38695d4dvib2f02nrd0hsn9-ninja-1.8.2/bin/ninja
building
make: *** No rule to make target 'built-sources'.  Stop.
builder for ‘/nix/store/881ggfjna4hbsffa1z1353r3fmk4zk8g-systemd-cryptsetup-generator.drv’ failed with exit code 2
error: build of ‘/nix/store/881ggfjna4hbsffa1z1353r3fmk4zk8g-systemd-cryptsetup-generator.drv’ failed
```

### Steps to reproduce

`nix-build -A systemd-cryptsetup-generator` on master

## Technical details

The systemd build system changed to meson but the systemd-cryptsetup-generator was not updated.

--- a/pkgs/os-specific/linux/systemd/cryptsetup-generator.nix
+++ b/pkgs/os-specific/linux/systemd/cryptsetup-generator.nix
@@ -10,21 +10,15 @@ stdenv.lib.overrideDerivation systemd (p: {
   outputs = [ "out" ];
 
   buildPhase = ''
-    make $makeFlags built-sources
-    make $makeFlags systemd-cryptsetup
-    make $makeFlags systemd-cryptsetup-generator
+    ninja systemd-cryptsetup systemd-cryptsetup-generator
   '';
 
-  # For some reason systemd-cryptsetup-generator is a wrapper-script
-  # with the current release of systemd. We want the real one.
-
-  # TODO: Remove `.libs` prefix when the wrapper-script is gone
   installPhase = ''
     mkdir -p $out/lib/systemd/
-    cp .libs/systemd-cryptsetup $out/lib/systemd/systemd-cryptsetup
-    cp .libs/*.so $out/lib/
+    cp systemd-cryptsetup $out/lib/systemd/systemd-cryptsetup
+    cp src/shared/*.so $out/lib/systemd/
 
     mkdir -p $out/lib/systemd/system-generators/
-    cp .libs/systemd-cryptsetup-generator $out/lib/systemd/system-generators/systemd-cryptsetup-generator
+    cp systemd-cryptsetup-generator $out/lib/systemd/system-generators/systemd-cryptsetup-generator
   '';
 })

